File: 3629-total-characters-in-string-after-transformations-i/total-characters-in-string-after-transformations-i.cpp
class Solution {
public:
    int M = 1e9+7;
    int lengthAfterTransformations(string s, int t) {
        vector<int> mp(26);
        for(char& c: s) mp[c-'a']++;
        for(int i=0; i<t; i++) {
            int z = mp[25];
            for(int j=25; j>0; j--) mp[j] = mp[j-1];
            mp[0] = z;
            mp[1] = (mp[1] + z) % M;
        }
        int res = 0;
        for (int& ele : mp) res = (res + ele) % M;
        return res;
    }
};

File: 3629-total-characters-in-string-after-transformations-i/total-characters-in-string-after-transformations-i.py
class Solution:
    def lengthAfterTransformations(self, s: str, t: int) -> int:
        count = [0] * 26
        for ch in s:
            count[ord(ch) - 97] += 1
        for _ in range(t):
            z = count[25]
            for j in range(25, 0, -1):
                count[j] = count[j-1]
            count[0] = z
            count[1] += z

        return sum(count) % 1000000007